bookkeeper-proto/src/main/proto/BookkeeperProtocol.proto

```diff
@@ -90,6 +90,7 @@ message ReadRequest {
     enum Flag {
         FENCE_LEDGER = 1;
         ENTRY_PIGGYBACK = 2;
+        RECOVERY_READ = 3;
```

Comment:
@merlimat @sijie 
There's a potential thorny BC issue here. On yahoo branch, RECOVERY_READ is 2.https://github.com/yahoo/bookkeeper/blob/165971209f933e0bcefaac997db8e75c8776f942/bookkeeper-server/src/main/proto/BookkeeperProtocol.proto#L83

@merlimat are any any the yahoo clients using protobufs? Or are they all on V2? If so it may not be a problem.

Reply:
I think the only clients using the v3 protocols are the auto-recovery daemons which are using the default settings. What would be the worst case if they do a read request with `ENTRY_PIGGYBACK` flag? 

The `RECOVERY_READ` flag is anyway optional, in the sense that is currently just used to signal urgency to the bookie

Reply:
a read request with `ENTRY_PIGGYBACK` is a normal read request. so it only means you don't assign priorities for recovery read, which I don't think it is a big deal.

Reply:
I think this definitely needs to be BC tested. The question is whether we block this change to do it, or just make it a blocker on the 4.7.0 release.

Reply:
my feeling is we need to do this as early as possible.  I am not sure if there is other similar changes coming up or not. I understand this is a port from yahoo branch, it probably doesn't make sense to block this change to be BC tested. I am fine with creating a blocker issue for adding BC tests for this change here.

Reply:
SGTM. Created issue #903